### PR TITLE
feat: automatic denom switcher

### PR DESCRIPTION
**Pull request by dni**
switches to sat denom of amount > x

closes https://github.com/BoltzExchange/boltz-web-app/issues/391

### CODE REVIEW

**Review by kilrau — CHANGES_REQUESTED** [review]
Second bullet point of https://github.com/BoltzExchange/boltz-web-app/issues/391 is missing: allow entering floats and autoswitch to btc if a `.` is entered. It needs both diretions, we had support cases the same way where people couldn't enter `0.1` as shown in sideswap.

**Comment by kilrau** [comment]
> autoswitch to btc if a `.` is entered

Still not working (doesn't let me enter `.` when on sats denom). We need this direction too.

**Comment by kilrau** [comment]
And conflicts need to be resolved

File: src/pages/Create.tsx [hunk 1]
```diff
@@ -60,10 +62,21 @@ import { enableWebln } from "../utils/webln";
 const Create = () => {
     let receiveAmountRef: HTMLInputElement, sendAmountRef: HTMLInputElement;
 
+    // if btc and amount > 9, switch to sat
+    // user failed to notice the non satoshi denomination
+    const changeDenomination = (amount: number) => {
+        if (denomination() === denominations.btc && amount > 9) {
+            setDenomination(denominations.sat);
+        } else if (denomination() === denominations.sat && amount < 1) {
```
Comment (michael1011) [inline src/pages/Create.tsx:70]: This has the side effect of switching to BTC denom when the amount is cleared. Questionable if we want that

Comment (kilrau) [reply src/pages/Create.tsx:70]: We don't. No switching on clearing.

Comment (dni) [reply src/pages/Create.tsx]: is it better now?  :)

**Review by kilrau — CHANGES_REQUESTED** [review]
Better but not there yet. when in sats deno and I enter `0.` it correctly switches to bitcoin and let's me enter. If I enter `1.` or `2.` etc it does weird things. 

[Screencast from 01-08-2024 07:28:14 PM.webm](https://github.com/BoltzExchange/boltz-web-app/assets/19181985/05682578-3f9a-4f75-9e82-801ae6eefd42)

**Review by kilrau — CHANGES_REQUESTED** [review]
My previous review comment is still not fixed. When I am in sat denomination and I enter `1.`  it jumps to `0.00000001`, should stay `1.`

**Review by michael1011 — APPROVED** [review]

**Pull request merged**